### DISCUSSION TITLE
Not able to close the model after saving the data from component

### DISCUSSION THREAD

**Discussion by MAdnanNaeem** [livewire/livewire#5334] [1. Help]
Hi there ! Please help me out to solve this issue in my code. 

**For Code  **

**App.blade.php**

 @livewireScripts
    <script type="text/javascript">
        window.livewire.on('closeNoteOrderModal',() => {
            $("#NoteOrderModal").modal('hide');
        }); </script>

**Order Component**

 NoteOrder::create([
            'encounter_id' => $this->visitId,
            'medicine' => $this->medicine,  ]);
        $this->resetInput();
        $this->toast('success', 'Order Added Successfully');
        $this->dispatchBrowserEvent('closeNoteOrderModal');

**Blade file**

                         
                                            {{-- Add new Order Modal --}}
                                            <div class="modal fade col-xs-12" id="NoteOrderModal" tabindex="-1"
                                                role="dialog" aria-labelledby="NoteOrderModalLabel"
                                                aria-hidden="true" wire:ignore.self>
                                                <div class="modal-dialog modal-dialog-centered" role="document">
                                                    <div class="modal-content">
                                                        {{-- header --}}
                                                        <div class="modal-header">
                                                            <h5 class="modal-title" id="NoteOrderModalLongTitle">
                                                                Order Details</h5>
                                                            <button type="button" class="close"
                                                                data-dismiss="modal" aria-label="Close">
                                                                <span aria-hidden="true">&times;</span>
                                                            </button>
                                                        <div class="modal-footer">
                                                            <button type="button" class="btn btn-secondary" data-dismiss="modal"
                                                                wire:click="resetInput()">Cancel</button>
                                                            <button type="submit" class="btn btn-primary"
                                                                wire:click.prevent="storeNoteOrder()">Save</button></div></div></div></div>

**Comment by ridxery** [comment]
You are listening Livewire Events on frontend but you are dispatching browser event from livewire.

if you change this 
`window.livewire.on('closeNoteOrderModal',() => {
        $("#NoteOrderModal").modal('hide');
    }); `

to this at your app.blade.php

`document.addEventListener('closeNoteOrderModal', () => {
        $("#NoteOrderModal").modal('hide');
    });` 

it should work, or use `$this->emit('closeNoteOrderModal')` on your livewire component controller

**Reply by MAdnanNaeem** [reply]
It still not worked for me. @ridxery

**Comment by polycode-nz** [comment]
As @ridxery  said.

Component:
`$this->dispatchBrowserEvent('closeNoteOrderModal');`

JS:
`window.addEventListener('closeNoteOrderModal', function(event)
{
    $('#NoteOrderModal').modal('hide');
});`
